core/controllers/base.py

```diff
@@ -406,6 +436,13 @@ def render_template(
             self.jinja2_env.get_template(filename).render(**values))
 
     def _render_exception(self, error_code, values):
+        """Renders an error page, or an error JSON response.
+
+         Args:
+            error_code: int. The HTTP status code (expected to be one of
+                        400, 401,404 or 500).
+            values: dict. list of values from JSON response.
```

Comment:
Edited. Cookies --> JSON response. From my understanding, the error_code is first asserted and then updated to 'code'  key in the JSON response. Please correct me if I miss something.